ansible_collections/f5networks/f5_modules/plugins/modules/bigip_device_group.py

```diff
@@ -298,6 +333,14 @@ def auto_sync(self):
             return 'enabled'
         return 'disabled'
 
+    @property
+    def asm_sync(self):
+        if self._values['asm_sync'] is None:
+            return None
+        elif self._values['asm_sync']:
+            return 'enabled'
+        return 'disabled'
+
```

Comment:
use flatten_boolean function here, and do something like
```
result = flatten_boolean(self._values['asm_sync'])
    if result == 'yes':
       return 'enabled
    if result == 'no':
      return 'disabled'

Reply:
I've mirrored the approach taken for all other API attributes, specifically `auto_sync`. `flatten_boolean` isn't a function currently used within the code which is why I think it would be better to stick to the existing approach. If this needs to be refactored later this could be done along with all other properties.

Reply:
you are going by wrong examples, flatten boolean is for ensuring that if i specify for example: true, yes, True etc. in the playbook we make it either 'yes' and 'no' same goes for whatever api returns, if we see enabled/on/True etc we ensure internally we have it as 'yes' or 'no' this is done for comparison. Look at other modules how we handle it.

Reply:
Refactoring to `flatten_boolean` seems like a good idea, other modules already use it, but this would be beyond the scope of this feature PR as it applies to all properties of class `UsableChanges`.

I think refactoring `flatten_boolean` should be considered for a separately.